src/balances/entities/schemas/balance.schema.ts

```diff
@@ -0,0 +1,44 @@
+import Ajv, { JSONSchemaType } from 'ajv';
+import { Balance } from '../balance.entity';
+import { TokenInfo } from '../../../common/entities/token-info.entity';
+import { TokenType } from '../../../common/entities/token-type.entity';
+
+const ajv = new Ajv({ coerceTypes: true });
+
+const tokenInfoSchema: JSONSchemaType<TokenInfo> = {
+  type: 'object',
+  properties: {
+    tokenType: {
+      type: 'string',
+      enum: [
+        TokenType.Erc20,
+        TokenType.Erc721,
+        TokenType.NativeToken,
+        TokenType.Unknown,
+      ],
+    },
+    address: { type: 'string' },
+    decimals: { type: 'number' },
+    symbol: { type: 'string' },
+    name: { type: 'string' },
+    logoUri: { type: 'string', nullable: true },
+  },
+  required: [],
+};
+
+ajv.addSchema(tokenInfoSchema, 'tokenInfo');
+
+const balanceSchema: JSONSchemaType<Balance> = {
+  type: 'object',
+  properties: {
+    tokenInfo: { $ref: 'tokenInfo' },
+    balance: { type: 'string' },
```

Comment:
@fmrsabino I'm not sure about this types. In the current CGW implementation `balance`, `fiatBalance` and `fiatConversion` are all `string`:
```
"balance": "25000000000000000",
"fiatBalance": "39.03450",
"fiatConversion": "1561.38"
```
But in this service we're returning on string and two numbers:
```
"balance": "25000000000000000",
"fiatBalance": 39.0503,
"fiatConversion": 1562.01
```

Do you have more context on this? I just created the JSONSchema taking the definition in `balance.entity.ts` but maybe it's wrong 🤔 I think this could affect clients expectations. For sure this could be addressed further, when we have the common DTOs in place, but just curious about it 🙂

Reply:
I think some of these fields were done due to how rational numbers were handled (internally by the service) and converting/forwarding them as string was what was established as part of the API contract.

In this case I'd say that we should keep them as `string` when serialising for the clients but keep it as a `number` on the DTO side as long as it doesn't loose precision on the value received.